pkg/kubectl/apply/visitor.go

```diff
@@ -34,6 +34,18 @@ type Strategy interface {
 
 	// MergeEmpty is invoked by EmptyElements when Merge is called
 	MergeEmpty(EmptyElement) (Result, error)
+
+	// Compare is invoked by element when conflict detection enabled
+	Compare
```

Comment:
here it is. It doesn't make sense for `Strategy` to embed `Compare`. The `Compare` method at the beginning should take this `Compare` interface, not `Strategy`